lib/gi/client.rb

```diff
@@ -1,46 +1,57 @@
 # frozen_string_literal: true
 
-require 'common/client/base'
-require 'gi/configuration'
-
 module GI
   # Core class responsible for api interface operations
   class Client < Common::Client::Base
     configuration GI::Configuration
 
     def get_institution_autocomplete_suggestions(params = {})
-      perform(:get, 'institutions/autocomplete', params, nil).body
+      response = perform(:get, 'institutions/autocomplete', params, nil)
```

Comment:
Looks like `perform` defaults the `headers` argument to `nil`, so you don't need to specify it in your call.
```suggestion
      response = perform(:get, 'institutions/autocomplete', params)
```
Applies to all the `perform` calls in this class.

Reply:
removed